apilib/model.py

```diff
@@ -39,6 +39,19 @@ def __init__(self, **kwargs):
                 raise exceptions.UnknownFieldException('Unknown field "%s"' % key)
             setattr(self, key, value)
 
+    def validate(self, error_context=None, context=None):
+        is_root = not error_context
+        error_context = error_context or ErrorContext()
+        context = self.make_parent_context(self.to_dict(), context) if context else None
+        for field_name, field in six.iteritems(self._field_name_to_field):
+            field._validate(getattr(self, field_name), error_context.extend(field=field_name), context)
+
+        if error_context.has_errors():
+            if is_root:
+                raise exceptions.ValidationError(error_context.all_errors())
+            return None
```

Comment:
Probably not important, but any reason not to return self here?

Reply:
Ah I see, this is copied from `from_json()`. I don't think you need to return anything from this method. If it's easier for chaining, you can always return `self`.

Reply:
yeah it's mostly just copied from `from_json()`. I actually didn't have it originally, but having it made the parallelism in the tests a lot easier. probably should add a comment to that effect?

Reply:
Can you just do `return self` without the other case? No comment needed, not atypical to do this for chaining.

Reply:
Hmm I'm not sure I understand. I did it this way because of the various `self.assertIsNone(m)` and `self.assertIsNone(m)` in e.g. `TestRequiredStringField`, just so I could loop more easily and change less test code. Primarily laziness! 

That said, I did just realize that I can wrap those asserts in `if use_json`, in which case I would return `self` from `validate`

Reply:
Yeah it made more sense to return `None` in `from_json` since that's a factory function, so if the inputs are invalid it can't return an object. Here you're just validating an object that already exists, so it's weird to return `None` sometimes and `self` sometimes. I'd either always return `self`, or don't return at all, and in either case update `run_test` to wrap the non-json case as needed.

Reply:
That makes sense. Will do.